src/main/scala/org/opennetworkinsight/dns/DNSPreLDA.scala

```diff
@@ -93,116 +78,57 @@ object DNSPreLDA {
         */
       val lines = Source.fromFile(feedbackFile).getLines().toArray.drop(1)
       val feedback: RDD[String] = sc.parallelize(lines)
-      val feedbackDataFrame = feedback.map(_.split(",")).map(row => Feedback(row(FrameTimeIndex),
+      feedback.map(_.split(","))
+        .filter(row => row(DnsSevIndex).trim.toInt == 3)
+        .map(row => Feedback(row(FrameTimeIndex),
         row(UnixTimeStampIndex),
         row(FrameLenIndex).trim.toInt,
         row(IpDstIndex),
         row(DnsQryNameIndex),
         row(DnsQryClassIndex),
         row(DnsQryTypeIndex),
         row(DnsQryRcodeIndex),
-        row(DnsSevIndex).trim.toInt)).toDF()
-      val result = feedbackDataFrame.filter("dnsSev = 3").select("frameTime", "unixTimeStamp", "frameLen", "ipDst",
-        "dnsQryName", "dnsQryClass", "dnsQryType", "dnsQryRcode").map(_.mkString(","))
-      val toDuplicate: org.apache.spark.rdd.RDD[String] = result.flatMap(x => List.fill(duplicationFactor)(x))
-      toDuplicate
+        row(DnsSevIndex).trim.toInt))
+        .flatMap(row => List.fill(duplicationFactor)(row))
+        .toDF()
+        .select("frameTime", "unixTimeStamp", "frameLen", "ipDst",
```

Comment:
I'm still thinking on what to do with those. 
Column names correspond to case class Feedback. If I want to use schema names I'd have to create Feedback case class with attributes like unix_tstamp instead of unixTimeStamp, frame_len instead of frameLen, etc. 

Best approach would be to implement schema for feedback, however given the near plans to restructure feedback I don't know if I should spend time on it. What do you think? 

`case class Feedback(frameTime: String,		   case class Feedback(frameTime: String,
                       unixTimeStamp: String,		                       unixTimeStamp: String,
                       frameLen: Int,		                       frameLen: Int,
                       ipDst: String,		                       ipDst: String,
                       dnsQryName: String,		                       dnsQryName: String,
                       dnsQryClass: String,		                       dnsQryClass: String,
                       dnsQryType: String,		                       dnsQryType: String,
                       dnsQryRcode: String,		                       dnsQryRcode: String,
                       dnsSev: Int) extends Serializable`